src/monitor/check_restart.py

```diff
@@ -0,0 +1,56 @@
+import time
+import subprocess
+from pathlib import Path
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+
+PROJECT_DIR = Path(__file__).resolve().parents[2]
+if __name__ == '__main__':
+    import sys
+    sys.path.append(str(PROJECT_DIR))
+	
+def get_last_line(filename):
+    with open(filename, 'r') as file:
+        lines = file.readlines()
+        if lines:
+            return lines[-1].strip()
+        else:
+            return None
+
+class OnMyWatch:
+	watchDirectory: Path = PROJECT_DIR / 'jobs/monitor_exec'
+    
+	def __init__(self):
+		self.observer = Observer()
+
+	def run(self):
+		event_handler = Handler()
+		self.observer.schedule(event_handler, self.watchDirectory, recursive = True)
+		self.observer.start()
+		try:
+			while True:
+				time.sleep(5)
+		except:
+			self.observer.stop()
+			print("Observer Stopped")
+
+		self.observer.join()
+
+
+class Handler(FileSystemEventHandler):
```

Comment:
There is no need for this class; it only has a staticmethod `on_any_event()`, so it is equivalent to using a function.